Client.java

```diff
@@ -3,11 +3,12 @@
 import Cinema;
 
 public interface Client {
-    public String username; //client username
+    public String username = null; //client username
+    public String password = null;
 
-    public boolean login(String username, String password);  //returns successful login or not, login also initializes state for client
+    public boolean login(CustomerController c) throws IOException;  //returns successful login or not, login also initializes state for client
```

Comment:
Since Client is an interface implemented by both Customer and Staff, we shouldn't stick to just a CustomerController approach. On that note however, it is probably a good idea to write a combined Controller for both Staff and Customer together. So rename your Controller as ClientController, and I will add in the required methods for the Staff CRUD as well.